akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -473,6 +473,177 @@ private[akka] final case class Conflate[In, Out](seed: In ⇒ Out, aggregate: (O
   override def restart(): Conflate[In, Out] = copy()
 }
 
+private[akka] sealed abstract class AbstractBatch[In, Out](max: Long, costFn: In ⇒ Long, seed: In ⇒ Out,
+                                                           aggregate: (Out, In) ⇒ Out, val in: Inlet[In],
+                                                           val out: Outlet[Out]) extends GraphStage[FlowShape[In, Out]] {
```

Comment:
it's not just the weight function:
- I overridden the `initialAttributes` in the concrete classes.
- If `Conflate` will be implemented in terms of `Batch`, we also need `max`
- most important (since the others could lie in the API function) is the received `Inlet` & `Outlet`. I want to pass these with a good name, and having an abstract class seemed to me the best option

so, should I change it? if so, what names should I supply the `Inlet`&`Outlet` in the abstract class with?